tests/jeos/firstrun.pm

```diff
@@ -58,6 +58,39 @@ sub verify_mounts {
     assert_script_run('mount -fva');
 }
 
+sub verify_hypervisor {
+    my $virt = script_output('systemd-detect-virt');
+
+    return 0 if (
+        is_qemu && $virt =~ /(qemu|kvm)/ ||
+        is_s390x && $virt =~ /zvm/ ||
+        is_hyperv && $virt =~ /microsoft/ ||
+        is_vmware && $virt =~ /vmware/);
+
+    die("Unknown hypervisor: $virt");
+}
+
+sub verify_norepos {
+    my $ret = script_run "zypper lr";
+
+    # Check ZYPPER_EXIT_NO_REPOS
+    die("Image should not contain any repos after first boot") if ($ret != 6);
+}
+
+sub verify_bsc {
+    if (is_qemu && script_run("rpm -q qemu-guest-agent") != 0) {
+        # Included in SLE-15-SP2+, TW and Leap
+        die("bsc#1207135 - Missing qemu-guest-agent from virtual images") unless is_sle('<15-SP2');
+    }
+
+    if (is_qemu && script_run("rpm -q grub2-x86_64-xen") == 0) {
+        die("bsc#1166474 - kvm-and-xen image contains grub2-x86_64-xe");
```

Comment:
I think this will also die on `aarch64` or `s390x` z/KVM image

Reply:
> I think this will also die on `aarch64` or `s390x` z/KVM image

But the package isn't supposed to be installed on those arches, right?